src/main/java/uk/gov/ons/ssdc/supporttool/schedule/RowChunkValidator.java

```diff
@@ -31,18 +33,41 @@ public RowChunkValidator(
   public boolean processChunk(Job job) {
     boolean hadErrors = false;
 
-    ColumnValidator[] columnValidators =
-        jobTypeHelper
-            .getJobTypeSettings(job.getJobType(), job.getCollectionExercise().getSurvey())
-            .getColumnValidators();
+    JobTypeSettings jobTypeSettings =
+        jobTypeHelper.getJobTypeSettings(job.getJobType(), job.getCollectionExercise());
 
     List<JobRow> jobRows =
         jobRowRepository.findTop500ByJobAndJobRowStatus(job, JobRowStatus.STAGED);
 
+    boolean getValidationRulesPerRow = false;
+    ColumnValidator[] columnValidators = null;
+
+    if (job.getJobType() == JobType.BULK_UPDATE_SAMPLE
+        || job.getJobType() == JobType.BULK_UPDATE_SAMPLE_SENSITIVE) {
+      getValidationRulesPerRow = true;
+    } else {
+      columnValidators = jobTypeSettings.getColumnValidators();
+    }
+
     for (JobRow jobRow : jobRows) {
       JobRowStatus rowStatus = JobRowStatus.VALIDATED_OK;
       List<String> rowValidationErrors = new LinkedList<>();
 
+      if (getValidationRulesPerRow) {
+        String fieldToUpdate = jobRow.getRowData().get("fieldToUpdate");
+        columnValidators =
+            jobTypeSettings.getColumnValidatorForSampleOrSensitiveDataRows(fieldToUpdate);
```

Comment:
calling this for every row will be very performance expensive. Should consider caching sooner rather than later?

Reply:
uses map already

Reply:
actually I'm wrong... the map will effectively cache it